src/mpi/coll/src/csel.c

```diff
@@ -643,14 +643,14 @@ static csel_node_s *prune_tree(csel_node_s * root, MPIR_Comm * comm_ptr)
                 else
                     node = node->success;
                 break;
-
+/*
```

Comment:
out of curiousity what is the eventual change to this block?

Reply:
Since we will no longer have subcomm, we will only remove hierarchy kind, and there is no point to do hierarchy kind check. We need check the current usages and rename the check. If it is checking for whether it's subgroup, we can remove the prune here; it has to be checked dynamically (on whether `coll_group` is `0`). If it is checking for topology, we should replace the check with something more specific.

Reply:
See the new commit 1bc0b28e05